### PR TITLE
Core ver. 8 transport seam migration

### PR DESCRIPTION
**Pull request by tmasternak**

### CODE REVIEW

File: src/NServiceBus.Transport.SqlServer/Addressing/EndpointAddressConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        /// <param name="schema">Custom schema value.</param>
+        public static void UseSchemaForEndpoint(this RoutingSettings settings, string endpointName, string schema)
+        {
+            Guard.AgainstNull(nameof(endpointName), endpointName);
+
+            var localEndpointName = settings.GetSettings().EndpointName();
+
+            if (endpointName.Equals(localEndpointName))
+            {
+                throw new ArgumentException("Custom schema cannot be specified for the local endpoint.");
```
Comment (SzymonPobiega) [inline src/NServiceBus.Transport.SqlServer/Addressing/EndpointAddressConfiguration.cs:26]: Should we point the user to the API that allows this? I mean the `DefaultSchema` API.

File: src/NServiceBus.Transport.SqlServer/Receiving/QueueCreator.cs [hunk 2]
```diff
@@ -11,74 +11,85 @@ namespace NServiceBus.Transport.SqlServer
...
+                    {
+                        await command.ExecuteNonQueryAsync().ConfigureAwait(false);
+
+                    }
+
+                    transaction.Commit();
                 }
             }
-            catch (Exception e) when (e.Message.StartsWith("There is already an object named") || e.Message.StartsWith("The operation failed because an index or statistics with name"))
+            catch (SqlException e) when (e.Number == 2714) //Object already exists
```
Comment (SzymonPobiega) [inline src/NServiceBus.Transport.SqlServer/Receiving/QueueCreator.cs:59]: can these two `catch` clauses be merged or are they handling different cases?

File: src/NServiceBus.Transport.SqlServer/SubscriptionTableCreator.cs [hunk 3]
```diff
@@ -21,19 +23,32 @@ public SubscriptionTableCreator(QualifiedSubscriptionTableName tableName, SqlCon
...
+                            CommandType = CommandType.Text
+                        })
+                        {
+                            await command.ExecuteNonQueryAsync().ConfigureAwait(false);
+                        }
+
+                        transaction.Commit();
+                    }
+                }
+                catch (SqlException e) when (e.Number == 2714) //Object already exists
```
Comment (SzymonPobiega) [inline src/NServiceBus.Transport.SqlServer/SubscriptionTableCreator.cs:46]: This code does not include the catch clause that is there in the regular queue creation (the one based on the exception message). Is this intentional?

**Review by SzymonPobiega — APPROVED** [review]
I have some minor comments/questions but generally it looks good 👍

File: src/NServiceBus.Transport.SqlServer/Receiving/QueueCreator.cs (hunk 2 above)
Comment (tmasternak) [reply src/NServiceBus.Transport.SqlServer/Receiving/QueueCreator.cs]: Merged. The two cases handled are: table exists (2714 error code) and index exists (1913 error) code.

File: src/NServiceBus.Transport.SqlServer/SubscriptionTableCreator.cs (hunk 3 above)
Comment (tmasternak) [reply src/NServiceBus.Transport.SqlServer/SubscriptionTableCreator.cs]: Subscription table does not define any indexes so 1913 code can't be throw. I've added the catch, either way, to be on the safe side if we ever add any indexes there.

File: src/NServiceBus.Transport.SqlServer/Addressing/EndpointAddressConfiguration.cs (hunk 1 above)
Comment (tmasternak) [reply src/NServiceBus.Transport.SqlServer/Addressing/EndpointAddressConfiguration.cs:26]: Fixed

**Pull request merged**